zero/gkyl_efit.h

```diff
@@ -26,28 +26,38 @@ struct gkyl_efit{
   double rmin, rmax, zmin, zmax;
 
   double psisep; // Separatrix psi for our DG representation
+  double psisep_cubic; // Separatrix psi for our cubic DG representation
                  // Can differ from sibry, but we need to keep sibry
                  // because fpol, q, etc. are defined based on it
 
-  struct gkyl_basis *rzbasis;
-  struct gkyl_rect_grid *rzgrid;
-  struct gkyl_range *rzlocal;
-  struct gkyl_range *rzlocal_ext;
+  struct gkyl_basis rzbasis;
+  struct gkyl_basis rzbasis_cubic;
+  struct gkyl_rect_grid rzgrid;
+  struct gkyl_range rzlocal;
+  struct gkyl_range rzlocal_ext;
+  struct gkyl_rect_grid rzgrid_cubic;
+  struct gkyl_range rzlocal_cubic;
+  struct gkyl_range rzlocal_cubic_ext;
   struct gkyl_array *psizr;
-  struct gkyl_array *psibyrzr;
-  struct gkyl_array *psibyr2zr;
+  struct gkyl_array *psizr_cubic;
+  struct gkyl_array *bmagzr;
+  struct gkyl_basis_ops_evalf *evf ;
 
-  struct gkyl_basis *fluxbasis;
-  struct gkyl_rect_grid *fluxgrid;
-  struct gkyl_range *fluxlocal;
-  struct gkyl_range *fluxlocal_ext;
+  struct gkyl_basis fluxbasis;
+  struct gkyl_rect_grid fluxgrid;
+  struct gkyl_range fluxlocal;
+  struct gkyl_range fluxlocal_ext;
   struct gkyl_array* fpolflux;
   struct gkyl_array* qflux;
 
   int num_xpts; // Number of X-points
   double *Rxpt; // R coordinates of X points
   double *Zxpt; // Z coordinates of X-points
 
+  int num_xpts_cubic; // Number of X-points of cubic rep
+  double *Rxpt_cubic; // R coordinates of X points of cubic rep
+  double *Zxpt_cubic; // Z coordinates of X-points of cubic rep
```

Comment:
Is there a reason you have separate cubic variables here? Why can't we just reuse Rxpt and Zxpt if you are making them double*? Can't we just make them the right size based on whether or not you're using cubics?

Reply:
The reason than Rxpt and Zxpt are double* is because there could be any number of X-points (single null, double null, stellarator with a bunch of Xpoints).

The reason I made separate cubic variables is that I wanted to store both. The efit updater knows nothing about field line tracing : it does not know whether the field line tracer will use the cubic or quadratic representation. So, the efit object stores the x-points of the cubic and quadratic representation just as it stores the cubic and quadratic representations of psi.

The field line tracer takes a user input of whether or not to use the cubics, so the tracer will decide whether to use the cubic rep of psi along with cubic X-points or the quadratic.

This should definitely get cleaner later when we speed up the root finders and only use the cubics. Then we can get rid of all the quadratic stuff. For now storing both seemed like a reasonable design to me.